mesonbuild/modules/i18n.py

```diff
@@ -166,6 +166,15 @@ def _get_data_dirs(state: 'ModuleState', dirs: T.Iterable[str]) -> T.List[str]:
     def merge_file(self, state: 'ModuleState', args: T.List['TYPE_var'], kwargs: 'MergeFile') -> ModuleReturnValue:
         if self.tools['msgfmt'] is None or not self.tools['msgfmt'].found():
             self.tools['msgfmt'] = state.find_program('msgfmt', for_machine=mesonlib.MachineChoice.BUILD)
+        if isinstance(self.tools['msgfmt'], ExternalProgram):
```

Comment:
If the default value for `self.tools['msgfmt']` was `NonExistingExternalProgram()`, you could simplify the logic by just using the `.found()` method...

Reply:
The complexity here came from the fact that sometimes we allow a program to be not found, but we need to know whether it's been searched for before. At the time it was easier to just use None. I can't remember offhand what find_program_impl returns for not-found programs...

Reply:
If it's not found we raise an error above, by calling find_program in "required: true" mode.

This if isinstance check is because the result can be a build.Executable and we cannot check its version, but also we don't care since it's assumed to work.